src/utils/hooks/useFetchProductCategories.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+import { API_BASE_URL } from '../constants';
+import { useLatestAPI } from './useLatestAPI';
+
+export function useFetchProductCategories() {
+  const { ref: apiRef, isLoading: isApiMetadataLoading } = useLatestAPI();
+  const [productCategories, setProductCategories] = useState(() => ({
+    categories: [],
+    isLoading: true,
+  }));
+
+  useEffect(() => {
+    if (!apiRef || isApiMetadataLoading) {
+      return () => {};
+    }
+
+    const controller = new AbortController();
+
+    async function getProductCategories() {
+      try {
+        setProductCategories({ categories: [], isLoading: true });
+
+        const response = await fetch(
+          `${API_BASE_URL}/documents/search?ref=${apiRef}&q=${encodeURIComponent(
+            '[[at(document.type, "category")]]'
+          )}&lang=en-us&pageSize=30`,
+          {
+            signal: controller.signal,
+          }
+        );
+        const data = await response.json();
+
+        const categories = data.results?.map(({ id, data: category }) => {
+          return { id, category };
+        });
+
+        setProductCategories({ categories, isLoading: false });
+      } catch (err) {
+        setProductCategories({ categories: [], isLoading: false });
+        console.error(err);
+      }
```

Comment:
Podemos cambiar al estructura de este cacho de código de la siguiente manera:

```jsx
try { } // si todo sale bien en esta parte del try obtenemos el valor del estado que nos interesa
catch { } // si salta algún error aqui lo llamamos `setError` con lo que salto del error
finally { } // si todo salió bien o hubo error, resulte lo que resulte aquí podemos bajar la bandera de loading
```

```suggestion
      try {
        const url = `${API_BASE_URL} /documents/search ? ref = ${apiRef}& q=${encodeURIComponent('[[at(document.type, "category")]]')}& lang=en - us & pageSize=30`;
        const response = await fetch(url, {
          signal: controller.signal,
        });
        const data = await response.json();

        const categories = data.results?.map(({ id, data: category }) => {
          return { id, category };
        });

        setCategories(categories);
      } catch (err) {
        console.error(err);
        setError(err);
      } finally {
        setLoading(false);
      }
```